diff --git a/src/Util.cpp b/src/Util.cpp
index 5806924..3052d18 100644
--- a/src/Util.cpp
+++ b/src/Util.cpp
@@ -2,12 +2,40 @@
 
 #include <regex>
 #include <string>
+#include <fstream>
 
 #include <sys/types.h>
 #include <unistd.h>
 #include <dirent.h>
 
 namespace ptimetracker {
+
+NEW_EXCEPTION_TYPE(ReadFileException)
+
+std::string readFile(std::string inFileName)
+{
+    //see https://stackoverflow.com/questions/18816126/c-read-the-whole-file-in-buffer
+    std::ifstream file(inFileName, std::ios::ate);
+    std::streamsize size = file.tellg();
+    file.seekg(0, std::ios::beg);
+
+    char* buffer = new char[size];
+
+
+    if(file.read(buffer, size))
+    {
+        delete[] buffer;
+        throw ReadFileException("Error while reading " + inFileName);
+    }
+    else 
+    {
+        std::string str(buffer);
+        delete[] buffer;
+        return str;
+    }
+}
+
+
 bool dirExists(std::string dirName)
 {
     DIR* dir = opendir(dirName.c_str());
